contracts/ethereum/src/PaymentRegistry.sol

```diff
@@ -75,23 +80,58 @@ contract PaymentRegistry is Initializable, OwnableUpgradeable, UUPSUpgradeable {
 
 //TODO change name to claimPaymentStarknet
     function claimPayment(uint256 orderId, uint256 destAddress, uint256 amount) external payable onlyOwnerOrMM {
-        bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount, Chain.Starknet));
-        TransferInfo storage transferInfo = transfers[index];
-        require(transferInfo.isUsed == true, "Transfer not found.");
+        _claimPaymentStarknet(orderId, destAddress, amount);
 
         uint256[] memory payload = new uint256[](5); //TODO why array of 256 if then filled with 128?
         payload[0] = uint128(orderId); // low
         payload[1] = uint128(orderId >> 128); // high
-        payload[2] = transferInfo.destAddress;
+        payload[2] = destAddress;
         payload[3] = uint128(amount); // low
         payload[4] = uint128(amount >> 128); // high
             
         _snMessaging.sendMessageToL2{value: msg.value}(
             StarknetEscrowAddress,
             StarknetEscrowClaimPaymentSelector,
             payload);
+    }
 
-        emit ClaimPayment(transferInfo);
+    function claimPaymentBatch(
+        uint256[] calldata orderIds,
+        uint256[] calldata destAddresses, 
+        uint256[] calldata amounts
+    ) external payable onlyOwnerOrMM() {
+        require(orderIds.length == destAddresses.length, "Invalid lengths.");
+        require(orderIds.length == amounts.length, "Invalid lengths.");
+
+        uint256[] memory payload = new uint256[](5 * orderIds.length + 1);
+
+        payload[0] = orderIds.length;
+        
+        for (uint32 idx = 0; idx < orderIds.length; idx++) {
+            uint256 orderId = orderIds[idx];
+            uint256 destAddress = destAddresses[idx];
+            uint256 amount = amounts[idx];
+
+            _claimPaymentStarknet(orderId, destAddress, amount);
+
+            uint32 base_idx = 1 + 5 * idx;
+            payload[base_idx] = uint128(orderId); // low
+            payload[base_idx + 1] = uint128(orderId >> 128); // high
+            payload[base_idx + 2] = destAddress;
+            payload[base_idx + 3] = uint128(amount); // low
+            payload[base_idx + 4] = uint128(amount >> 128); // high
+        }
+        
+        _snMessaging.sendMessageToL2{value: msg.value}(
+            StarknetEscrowAddress,
+            StarknetEscrowClaimPaymentBatchSelector,
+            payload);
+    }
+
+    function _claimPaymentStarknet(uint256 orderId, uint256 destAddress, uint256 amount) internal view {
+        bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount, Chain.Starknet));
+        TransferInfo storage transferInfo = transfers[index];
+        require(transferInfo.isUsed == true, "Transfer not found.");
     }
```

Comment:
I think this function can be refactored to a modifier, and be used as `modifier onlyOwnerOrMM` is used in this same contract

Reply:
Unfortunately this is not possible, since it is used in claimPaymentBatch function as well